pkgs/applications/networking/geph/default.nix

```diff
@@ -0,0 +1,127 @@
+{ lib
+, stdenvNoCC
+, rustPlatform
+, fetchFromGitHub
+, buildNpmPackage
+, perl
+, pkg-config
+, glib
+, webkitgtk
+, libappindicator-gtk3
+, libayatana-appindicator
+, cairo
+, openssl
+}:
+
+let
+  geph-meta = with lib; {
+    description = "A modular Internet censorship circumvention system designed specifically to deal with national filtering.";
+    homepage = "https://geph.io";
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ penalty1083 ];
+  };
+in
+{
+  cli = rustPlatform.buildRustPackage rec {
+    pname = "geph4-client";
+    version = "4.7.0-beta.4";
+
+    src = fetchFromGitHub {
+      owner = "geph-official";
+      repo = pname;
+      rev = "v${version}";
+      hash = "sha256-elPZqkrh++1hoYpmhDJlBLmr0ELVUESkRAj+mu8UkuU=";
+    };
+
+    cargoHash = "sha256-D4o9i+vjYgIcG3YslAck19uJsglsGaO6MLXdmsTSnVw=";
+
+    nativeBuildInputs = [ perl ];
+
+    meta = geph-meta // {
+      license = with lib.licenses; [ gpl3Only ];
+    };
+  };
+
+  gui = stdenvNoCC.mkDerivation rec {
+    pname = "geph-gui";
+    version = "4.7.0-beta.4";
+
+    src = fetchFromGitHub {
+      owner = "geph-official";
+      repo = "gephgui-pkg";
+      rev = "6439945d6baa4b2a43bed7fa5a55ddbfcfc099be";
+      hash = "sha256-0cisnOJFaZHMb+KXdV5pbAayegtDPUHSWG/NMAuQJ4A=";
+      fetchSubmodules = true;
+    };
+
+    gephgui = buildNpmPackage {
+      pname = "gephgui";
+      inherit version src;
+
+      sourceRoot = "source/gephgui-wry/gephgui";
+
+      postPatch = ''
+        rm package-lock.json
+        ln -s ${./package-lock.json} ./package-lock.json
```

Comment:
Why do we need to delete the file?

Reply:
> I generated a lock file because fetch-npm-deps doesn't work with the upstream lock file. It seems fetch-npm-deps ignores deps without integrity.

We need to generate a correct lock file.

Reply:
It seems the problem is fixed upstream so this is not needed anymore.

Reply:
They removed the lock file in 4.7.8 so I have to generate one again.